### PR TITLE
622 refactor top navbar

### PR DESCRIPTION
**Pull request by kasugaijin**
# 🔗 Issue
https://github.com/rubyforgood/pet-rescue/issues/622

# ✍️ Description
Refactors top nav to replace links with relevant links and remove avatar and place the avatar dropdown links in navbar. 
Also fixes the error we were getting if a signed in user tries to access sign up page.

https://www.loom.com/share/8b171d6bc07f417e890e615f39c68cca?sid=db9f76dc-bb09-4448-9d76-6721aa1f0e87

# 📷 Screenshots/Demos

### CODE REVIEW

File: app/views/layouts/shared/_navbar.html.erb [hunk 1]
```diff
@@ -19,75 +19,39 @@
...
         <% if user_signed_in? %>
-          <li class="nav-item d-block d-lg-none">
-            <%= button_to 'Log Out', destroy_user_session_path, method: :delete, data: { turbo: false }, class: 'nav-link' %>
-          </li>
+          <% if current_user.adopter_foster_account %>
+            <li class='nav-item'>
+              <%= active_link_to 'Dashboard', adopter_foster_dashboard_index_path, class: 'nav-link' %>
+            </li>
+          <% end %>
+          <% if current_user.staff_account && !current_user.staff_account.deactivated? %>
```
Comment (mononoken) [inline app/views/layouts/shared/_navbar.html.erb:50]: An alternative approach would be to utilize ActionPolicy here. That would keep authz logic out of the view. It usually works great for using policies to check logic like this because it can infer the correct policy from the record passed. However, since `dashboard` is not a record, it's a bit wordy in this case:
```ruby
<% if allowed_to?(:index?, with: Organizations::DashboardPolicy, context: {organization: Current.organization}) %>

```
This makes it so instead of writing a conditional for the user having the right attributes, we check simply if the user passes the policy rule and let the policy handle the logic.

File: app/views/layouts/shared/_navbar.html.erb [hunk 2]
```diff
@@ -19,75 +19,39 @@
...
+          <%= active_link_to 'Contact', new_contact_path, class: 'nav-link' %>
         </li>
         <li class="nav-item">
           <%= active_link_to 'Donate', donate_path, class: 'nav-link' %>
         </li>
         <% if user_signed_in? %>
-          <li class="nav-item d-block d-lg-none">
-            <%= button_to 'Log Out', destroy_user_session_path, method: :delete, data: { turbo: false }, class: 'nav-link' %>
-          </li>
+          <% if current_user.adopter_foster_account %>
```
Comment (mononoken) [inline app/views/layouts/shared/_navbar.html.erb:45]: See https://github.com/rubyforgood/pet-rescue/pull/642/files#r1564502741

In this case it would be:
```ruby
<% if allowed_to?(:index?, with: Organizations::AdopterFosterDashboardPolicy, context: {organization: Current.organization}) %>
```

**Review by mononoken — APPROVED** [review]
Looks good! Good catch on the devise method. Thanks, @kasugaijin !

File: app/views/layouts/shared/_navbar.html.erb (hunk 2 above)
Comment (kasugaijin) [reply app/views/layouts/shared/_navbar.html.erb]: Ah yes good call. I made these two changes!
